underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystem.java

```diff
@@ -126,6 +126,22 @@ public GCSUnderFileSystem(AlluxioURI uri) throws ServiceException {
     mBucketMode = GCSUtils.translateBucketAcl(acl, mAccountOwnerId);
   }
 
+  protected GCSUnderFileSystem(AlluxioURI uri,
```

Comment:
ah, only static methods can be called before the class is constructed. Anyway, I think the second approach is better and we're in a good place now